### ISSUE TITLE
[python] python packages for opensuse tumbleweed renamed

### ISSUE THREAD

**Issue by aboe76** [saltstack-formulas/libvirt-formula#82]
### Formula commit hash / release tag

libvirt-formula master branch

### Versions reports (master & minion)

```
Salt Version:
Salt: 3002.2

Dependency Versions:
cffi: 1.14.4
cherrypy: unknown
dateutil: 2.8.1
docker-py: Not Installed
gitdb: Not Installed
gitpython: Not Installed
Jinja2: 2.11.2
libgit2: 1.1.0
M2Crypto: 0.37.1
Mako: 1.1.3
msgpack: 1.0.2
msgpack-pure: Not Installed
mysql-python: 1.4.6
pycparser: 2.20
pycrypto: 3.9.9
pycryptodome: 3.9.9
pygit2: 1.3.0
Python: 3.8.6 (default, Nov 09 2020, 12:09:06) [GCC]
python-gnupg: 0.4.6
PyYAML: 5.3.1
PyZMQ: 20.0.0
smmap: 3.0.4
timelib: 0.2.4
Tornado: 4.5.3
ZMQ: 4.3.3

System Versions:
dist: opensuse-tumbleweed 20210118 n/a
locale: utf-8
machine: x86_64
release: 5.10.7-1-default
system: Linux
version: openSUSE Tumbleweed 20210118 n/a
```

### Pillar / config used

```
*redacted*:
    ----------
    libvirt:
        ----------
        daemon_opts:
            ----------
            LIBVIRTD_ARGS:
                --timeout 120
            LIBVIRTD_CONFIG:
                /etc/libvirt/libvirtd.conf
            LIBVIRTD_NOFILES_LIMIT:
                2048
        libvirtd:
            ----------
            auth_unix_rw:
                none
            unix_sock_group:
                libvirt
            unix_sock_rw_perms:
                0770
        lookup:
            ----------
            libvirt_service:
                libvirtd-admin.socket
            python3_pkg:
                python38-libvirt-python
```
---

### Describe the bug

opensuse tumbleweed just renamed all of its python3 packages:
correct python3 package: python38-libvirt-python

### Expected behaviour

add a parameter yaml file for tumbleweed with the correct default package:
```
os_family: Suse
oscodename: openSUSE Tumbleweed
osfinger: openSUSE Tumbleweed-20210118
osfullname: openSUSE Tumbleweed
```

### Attempts to fix the bug

added the correct package lookup in pillar
```
        lookup:
            ----------
            python3_pkg:
                python38-libvirt-python
```

**Comment by baby-gnu** [comment]
Thanks a lot @aboe76

**Comment by baby-gnu** [comment]
To fix the bug, you could provide the file `parameters/osfinger/openSUSE Tumbleweed-20210118.yaml` with the following content:

```yaml
# -*- coding: utf-8 -*-
# vim: ft=yaml
#
# Setup variables specific to salt['config.get']('osfinger') == openSUSE Tumbleweed-20210118.
# You just need to add the key:values for this `osfinger` that differ
# from `defaults.yaml`.
#
# If you do not need to provide defaults via the `osfinger` config,
# you can remove this file or provide at least an empty dict, e.g.
# values: {}
---
values:
  python2_pkg: ~
  python3_pkg: python38-libvirt-python
...
```

I'll submit a PR to add that file.

**Comment by myii** [comment]
> To fix the bug, you could provide the file `parameters/osfinger/openSUSE Tumbleweed-20210118.yaml` with the following content:

@baby-gnu We may have to rethink the naming of that file, since that `20210118` value is probably going to change every day!

@aboe76 I thought I'd try to build some Tumbleweed pre-salted images but `salt-bootstrap` is failing because the base Docker image doesn't have `/bin/systemctl`.

https://github.com/saltstack/salt-bootstrap/blob/18e906334997a37047f9699705e855990a1fb96b/bootstrap-salt.sh#L6541

```bash
        if [ -f /bin/systemctl ]; then
```

Any ideas of how this could be fixed?

**Comment by baby-gnu** [comment]
> @baby-gnu We may have to rethink the naming of that file, since that `20210118` value is probably going to change every day!

Ok, we can use `oscodename` for that case, no?

**Comment by myii** [comment]
> Ok, we can use `oscodename` for that case, no?

@baby-gnu I wanted to get some pre-salted images built so that I could investigate the platform a bit further.  I may just hack the bootstrap script in order to do that.  I'll let you know if I get anywhere.

**Comment by myii** [comment]
@baby-gnu Got the grains:

grain | value
-- | --
os | SUSE
os_family | Suse
oscodename | openSUSE Tumbleweed
osfinger | openSUSE Tumbleweed-20210118
osfullname | openSUSE Tumbleweed
osmajorrelease | 20210118
osrelease | '20210118’
osrelease_info | [20210118]

So `oscodename` it is.  And from a verification perspective, we're going to need to treat like any other rolling release, so either using the `-latest` suffix, or simply leaving it as `opensuse-tumbleweed.yaml`.

**Comment by baby-gnu** [comment]
> Got the grains:

Thanks @myii, @aboe76 put some in the initial ticket.

**Comment by myii** [comment]
> ```shell
>         if [ -f /bin/systemctl ]; then
> ```
> 
> Any ideas of how this could be resolved?

@aboe76 Don't worry, I've figured it out.  Need to ensure that the `systemd` package is installed and the file to be checked is `/usr/bin/systemctl`.  Planning to put through a PR to the `salt-bootstrap`.

**Comment by myii** [comment]
@aboe76 @baby-gnu Once this is merged, we can get pre-salted images built for Tumbleweed:

* https://github.com/saltstack/salt-bootstrap/pull/1525

**Comment by baby-gnu** [comment]
Closed with #83
